mod.d/enadiag.yaml

```diff
@@ -0,0 +1,270 @@
+          self.wd_expired = wd_expired
+          self.interface_down = interface_down
+          self.admin_q_pause = admin_q_pause
+          self.bw_in_allowance_exceeded = bw_in_allowance_exceeded
+          self.bw_out_allowance_exceeded = bw_out_allowance_exceeded
+          self.pps_allowance_exceeded = pps_allowance_exceeded
+          self.conntrack_allowance_exceed = conntrack_allowance_exceeded
+          self.linklocal_allowance_exceeded = linklocal_allowance_exceeded
+
+  def get_interfaces():
+      """
+      Return a list of non-loopback interfaces on the system.
+
+      Returns:
+          interface_list (list): List of interfaces (e.g. ["eth0", "eth1"]
+      """
+      try:
+          interface_list = []
+          # Get all net interfaces except loopback, append to list, sort it.
+          for interface in os.listdir("/sys/class/net"):
+              if interface != "lo":
+                  interface_list.append(interface)
+          if len(interface_list) > 0:
+              interface_list = sorted(interface_list)
+          # Return an error if the list is somehow empty (didn't find any network interfaces)
+          else:
+              raise ValueError("No interfaces found.")
+      # Catch OSError in Python2 and FileNotFoundError in Python3 (inherits from OSError)
+      except OSError:
+          print("Unable to build interface list.")
+      return interface_list
+
+  # Create list with get_interfaces and create Interface class objects with them.
+  interfaces = get_interfaces()
+  interface_results = [Interface(name) for name in interfaces]
+
+  def get_ena_version():
+      """
+      Return the version of the driver for the first alphabetically ordered non-virtual network interface.
+
+      Returns:
+          driver_version (str): version os driver (e.g. 2.2.4g)
+      """
+      try:
+          interface_list = []
+          # Add all non-virtual interfaces to the list and sort it
+          for interface in os.listdir("/sys/class/net"):
+              if "virtual" not in os.path.abspath(os.readlink("/sys/class/net/{}".format(interface))):
+                  interface_list.append(interface)
+          if len(interface_list) > 0:
+              interface_list = sorted(interface_list)
+              # Open the file, strip out newline characters
+              with open ("/sys/class/net/{}/device/driver/module/version".format(interface_list[0]), "r") as file:
+                  driver_version = file.read().replace('\n', '')
+          # Return an error if the list is somehow empty (didn't find any network interfaces)
+          else:
+              driver_version = "Unknown"
+      # Catch OSError in Python2 and FileNotFoundError in Python3 (inherits from OSError)
+      except OSError:
+          driver_version = "Unknown"
+      return driver_version
+
+  def compare_ena_version():
+      """
+      Check to see if the ena version is new enough to have the necessary stats available.
```

Comment:
I think this means the driver is new enough to support some certain stats via ethtool. Could you clarify?
https://github.com/amzn/amzn-drivers/blob/master/kernel/linux/ena/RELEASENOTES.md

Reply:
ethtool is how we pull them in this module, but they are theoretically available via other mechanisms as well. I feel like I am pretty clear that this availability is based on the driver version?

Reply:
Right, but why 2.2.0?

Reply:
Cleared up over chat, I was misunderstanding the ask. Will fix.